modules/develop/pages/consume-data/consumer-offsets.adoc

```diff
@@ -4,20 +4,125 @@
+
+NOTE: More advanced consumers can read data from Redpanda without using a consumer group, by requesting to read a specific topic + partition + offset range. This pattern is often used by stream processing systems such as Apache Spark and Apache Flink, which have their own mechanisms for assigning work (/partitions) to consumers.
+
+When the group coordinator receives an OffsetCommitRequest, it appends the request to the https://kafka.apache.org/documentation/#compaction[compacted^] Kafka topic `pass:[__consumer_offsets]`.
 The broker sends a successful offset commit response to the consumer only after all the replicas of the offsets topic receive the offsets.
 If the offsets fail to replicate within a configurable timeout, the offset commit fails and the consumer may retry the commit after backing off.
-The brokers periodically compact the offsets topic, because it only needs to maintain the most recent offset commit for each partition.
+The brokers periodically compact the `pass:[__consumer_offsets]` topic, because it only needs to maintain the most recent offset commit for each partition.
 The coordinator also caches the offsets in an in-memory table to serve offset fetches quickly.
 
-When stored in a partition, a sequential number is assigned to each event for order. This number is the _offset_. The offset allows consumer event processing to start or continue from where it left off. So, if you have a consumer application outage that prevents your application from receiving events, you can use the consumer offset to select and send only the events that occurred when the application was down. By default, the offset starts at 0, but you can manually specify a specific start value.
+== Commit strategies
+
+There are multiple commit strategies available. The most common follow.
+
+=== Auto commit
+
+Auto commit is the default commit strategy, where the client automatically commits offsets at regular intervals. You can enable this by setting `enable.auto.commit` to `true`. The client will then commit offsets every `auto.commit.interval.ms` milliseconds.
+
+The primary advantage of the auto commit approach is its simplicity. Once configured, the consumer requires no additional effort, as commits are managed in the background. However, this "hands-off" approach has a drawback: the consumer is unaware of what was committed or when. As a result, after an application restart, some messages may be reprocessed (since consumption resumes from the last committed offset, which may include already processed messages). The behavior provides an “at least once” delivery guarantee.
```

Comment:
```suggestion
The primary advantage of the auto commit approach is its simplicity. After it is configured, the consumer requires no additional effort. Commits are managed in the background. However, the consumer is unaware of what was committed or when. As a result, after an application restart, some messages may be reprocessed (since consumption resumes from the last committed offset, which may include already processed messages). The strategy guarantees at-least-once delivery.
```